### ISSUE TITLE
systemd-resolved listen to 0.0.0.0:5355

### ISSUE THREAD

**Issue by lclgo** [systemd/systemd#18155]
**systemd version the issue has been seen with**
 > systemd-245

**Used distribution**
 > Fedora 32

**Linux kernel version used** (`uname -a`)
 > Linux localhost.localdomain 5.6.6-300.fc32.aarch64

**CPU architecture issue was seen on**
 > aarch64

**Expected behaviour you didn't see**
 > systemd-resolved listens to 224.0.0.252:5355

**Unexpected behaviour you saw**
 > systemd-resolved listens to 0.0.0.0:5355

**Steps to reproduce the problem**
 > systemctl restart systemd-resolved; netstat -nap | grep systemd-resolved

**Comment by Geass-LL** [comment]
According to [https://tools.ietf.org/html/rfc4795](https://tools.ietf.org/html/rfc4795), I think systemd-resolved should listen to 244.0.0.252 and FF02:0:0:0:0:0:1:3. Don't understandard why systemd-resolved listen to 0.0.0.0 and ::: .
> LLMNR queries are sent to and received on port 5355.  The IPv4 link-
   scope multicast address a given responder listens to, and to which a
   sender sends queries, is 224.0.0.252.  The IPv6 link-scope multicast
   address a given responder listens to, and to which a sender sends all
   queries, is FF02:0:0:0:0:0:1:3.

**Comment by poettering** [comment]
I am sorry, but this is really a case where I can just say: RTFM! 

It's explained in the very first section of the man page.

https://www.freedesktop.org/software/systemd/man/systemd-resolved.service.html#Description

Please, try to keep the noise level down in this bug tracker. If you have a question and not a bug or feature request: first thing: check the documentation, then contact a suitable forum. But please, don't post such rubbish questions here. Thank you for understanding.

**Comment by GoodMirek** [comment]
@Geass-LL Have you ever been able to find why systemd-resolved listen to 0.0.0.0:5355 ?
I cannot find anything in the doc referred by @poettering explaining that.

**Comment by GoodMirek** [comment]
I believe the explanation is in [RFC4795, section 2.3](<https://datatracker.ietf.org/doc/html/rfc4795#section-2.3>):

```
   In responding to queries:

   (a)  Responders MUST listen on UDP port 5355 on the link-scope
        multicast address(es) defined in Section 2, and on TCP port 5355
        on the unicast address(es) that could be set as the source
        address(es) when the responder responds to the LLMNR query.
```
Listening on unicast address, TCP port 5355 is the required behavior.

**Comment by Geass-LL** [comment]
> @Geass-LL Have you ever been able to find why systemd-resolved listen to 0.0.0.0:5355 ? I cannot find anything in the doc referred by @poettering explaining that.

Maybe this line: "Additionally, systemd-resolved provides a local DNS stub listener on IP address 127.0.0.53 on the local loopback interface. Programs issuing DNS requests directly, bypassing any local API may be directed to this stub, in order to connect them to systemd-resolved. Note however that it is strongly recommended that local programs use the glibc NSS or bus APIs instead (as described above), as various network resolution concepts (such as link-local addressing, or LLMNR Unicode domains) cannot be mapped to the unicast DNS protocol." in https://www.freedesktop.org/software/systemd/man/systemd-resolved.service.html#Description

I don't know much about DNS, but listening to 0.0.0.0 should be something reletaed to `broadcast` ? And I didn't find anything about `broadcast` in RFC4795. So I believe systemd may have its own implementation which differs to RFC4795.
